cdap-common/src/test/java/io/cdap/cdap/common/internal/remote/RemoteTaskExecutorTest.java

```diff
@@ -88,16 +82,85 @@ public void modify(ChannelPipeline pipeline) {
 
   @Before
   public void beforeTest() {
-    published = new ArrayList<>();
-    mockMetricsCollector = new AggregatedMetricsCollectionService(1000L) {
+    metricCollectors = new HashMap<>();
+    mockMetricsCollector = createMockMetricsCollectionService();
+    mockMetricsCollector.startAndWait();
+    registered = discoveryService.register(URIScheme.createDiscoverable(Constants.Service.TASK_WORKER, httpService));
+  }
+
+  private MetricsCollectionService createMockMetricsCollectionService() {
+    return new MetricsCollectionService() {
```

Comment:
Create anonymous `MetricsCollectionService` instance to replace `AggregatedMetricsCollectionService`, which is from cdap-watchdog and causes circular dependency issue. The anonymous `MetricsCollectionService` simply collects metrics in a map